public/js/invertedIndex.js

```diff
@@ -0,0 +1,189 @@
+/**
+ * @class InvertedIndex
+ */
+class InvertedIndex {
+  /**
+   * Creates an instance of InvertedIndex.
+   * @param {Object} fileUploaded
+   * @memberOf InvertedIndex
+   */
+  constructor(fileUploaded) {
+    this.uploadedFile = fileUploaded;
+  }
+  /**
+   * logFile (returns the file that has been uploaded)
+   * @returns {Object} - returns the file that has been uploaded
+   * @memberOf InvertedIndex
+   */
+  logFile() {
+    return this.uploadedFile;
+  }
+  /**
+   * checkFileExtension (it returns the file extension)
+   * @param {string} uploadedFileName
+   * @return {string} - returns a string of the file extension of the uploaded file
+   * @memberOf InvertedIndex
+   */
+  checkFileExtension(uploadedFileName) {
+    const array = uploadedFileName.split('.');
+    return array[array.length - 1];
+  }
+  /**
+   * checkIfJson (it returns true if the file extension is .json)
+   * @param {string} uploadedFileName
+   * @returns {boolean} - returns true if the file extension is json else false
+   * @memberOf InvertedIndex
+   */
+  checkIfJson(uploadedFileName) {
+    const fileExtension = this.checkFileExtension(uploadedFileName);
+    return fileExtension.toLowerCase() === 'json';
+  }
+  /**
+   * containsTitleText (it checks if the content of the file contains title and text)
+   * @returns {Boolean} - returns true if the file contains title and text or false if it does not
+   * @memberOf InvertedIndex
+   */
+  containsTitleText() {
+    for (let arrayIndex = 0; arrayIndex < this.uploadedFile.length; arrayIndex += 1) {
+      if (('title' in this.uploadedFile[arrayIndex]) && ('text' in this.uploadedFile[arrayIndex])) {
+        return true;
+      }
+    }
+    return false;
+  }
+  /**
+   * getTitlesAndTexts (it gets all the titles and texts in the uploaded file)
+   * @returns {Object} - returns an object with keys titles and texts
+   * @memberOf InvertedIndex
+   */
+  getTitlesAndTexts() {
```

Comment:
Nitpick and question! Wouldn't it be more straightforward to just return an object like `{ titles: [], texts: [] }` instead of the array of objects `[{ titles: [] }, { texts: [] }]`?